src/collector/otlp/service.rs

```diff
@@ -203,21 +204,57 @@ fn ingest_locations(dic: &ProfilesDictionary) -> Result<Vec<Frame>, Status> {
             return Err(Status::invalid_argument("mapping index is out of bounds"));
         };
 
-        let build_id = get_attr(
-            atab,
-            mapping.attribute_indices.to_vec(),
-            "process.executable.build_id.htlhash", // OTel Profiling specific build ID.
-        )
-        .or_else(|_| {
-            get_attr(
+        let build_id;
+        let generated_build_id;
+        let build_id_str = if !mapping.attribute_indices.is_empty() {
+            build_id = get_attr(
                 atab,
                 mapping.attribute_indices.to_vec(),
-                "process.executable.build_id.profiling", // Legacy OTel Profiling specific build ID.
+                "process.executable.build_id.htlhash", // OTel Profiling specific build ID.
             )
-        })?;
+            .or_else(|_| {
+                get_attr(
+                    atab,
+                    mapping.attribute_indices.to_vec(),
+                    "process.executable.build_id.profiling", // Legacy OTel Profiling specific build ID.
+                )
+            })?;
+            build_id
+        } else {
+            // Fallback option: Generate xxh3 hash over all fields of all loc.line elements
+            // if there is no build_id attribute.
+            let mut hasher = xxh3::Xxh3::new();
+            for line in &loc.line {
+                if line.function_index != 0 {
+                    if let Some(fn_ref) = ftab.get(line.function_index as usize) {
+                        // Hash function name if available
+                        if let Ok(Some(function_name)) =
+                            get_str_opt(stab, fn_ref.name_strindex as usize, "function name")
+                        {
+                            hasher.update(function_name.as_bytes());
+                        }
+                        // Hash function filename if available
+                        if let Ok(Some(file_name)) = get_str_opt(
+                            stab,
+                            fn_ref.filename_strindex as usize,
+                            "function filename",
+                        ) {
+                            hasher.update(file_name.as_bytes());
+                        }
+                    }
+                } else {
+                    // If function_index is 0, hash the index itself as fallback.
+                    hasher.update(&line.function_index.to_le_bytes());
```

Comment:
How does this make any difference? Since function_index is always 0 here, it just adds a constant to the hash. I think it is superfluous.

Reply:
removed it